needy/__main__.py

```diff
@@ -196,15 +202,96 @@ def development_mode(args=[]):
         help='if given, will return 0 if dev-mode is enabled, or 1 otherwise')
     parameters = parser.parse_args(args)
 
-    with LocalConfiguration(os.path.join(Needy.resolve_needs_directory('.'), 'config.json')) as local_configuration:
-        needy = Needy('.', parameters, local_configuration=local_configuration)
+    with __configured_needy(parameters) as needy:
         if parameters.query:
             return 0 if needy.development_mode(parameters.library) else 1
         needy.set_development_mode(parameters.library, not parameters.disable)
 
     return 0
 
 
+def cache_path(args=[]):
+    parser = argparse.ArgumentParser(
+        prog='%s cache-path' % os.path.basename(sys.argv[0]),
+        description='Configures a cache for satisfied needs.',
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        epilog="""available commands:
+  set    sets the cache path
+  unset  unsets the cache path
+  query  emits the current cache path
+
+Use '%s cache-path <command> --help' to get help for a specific command.
+""" % os.path.basename(sys.argv[0])
+    )
+    parser.add_argument('command', help='see below')
+    parser.add_argument('args', nargs=argparse.REMAINDER)
+    parameters = parser.parse_args(args)
+
+    def set_cache_path(args=[]):
+        needs_dir = Needy.resolve_needs_directory('.')
+        default_cache_path = os.path.join(needs_dir, '.cache') if needs_dir else None
+        parser = argparse.ArgumentParser(
+            prog='%s cache-path set' % os.path.basename(sys.argv[0]),
+            description='set a cache path.',
+            formatter_class=argparse.RawDescriptionHelpFormatter)
+        parser.add_argument(
+            'path',
+            nargs='?',
+            default=default_cache_path,
+            help='new cache path')
+        parameters = parser.parse_args(args)
+
+        with __configured_needy(parameters) as needy:
+            configuration = {
+                'path': os.path.realpath(parameters.path),
+            }
+            needy.set_cache(Directory(configuration))
```

Comment:
i'd prefer this to use more typical initialization: `Directory(path=parameters.path)`. it's generally more legible and usable. for serialization / deserialization, there are other ways that don't sacrifice that (e.g. pickle?)